ksql-engine/src/test/java/io/confluent/ksql/topic/TopicDeleteInjectorTest.java

```diff
@@ -0,0 +1,184 @@
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.ksql.metastore.MutableMetaStore;
+import io.confluent.ksql.metastore.model.StructuredDataSource;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.parser.tree.DropStream;
+import io.confluent.ksql.parser.tree.ListProperties;
+import io.confluent.ksql.parser.tree.QualifiedName;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.serde.DataSource.DataSourceSerDe;
+import io.confluent.ksql.services.FakeKafkaTopicClient;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlConstants;
+import io.confluent.ksql.util.KsqlException;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.TopicPartitionInfo;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TopicDeleteInjectorTest {
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  @Mock
+  private MutableMetaStore metaStore;
+  @Mock
+  private StructuredDataSource<?> source;
+  @Mock
+  private SchemaRegistryClient registryClient;
+  @Mock
+  private FakeKafkaTopicClient topicClient;
+
+  private TopicDeleteInjector deleteInjector;
+  private ConfiguredStatement<DropStream> dropStatement;
+
+  @Before
+  public void setUp() {
+    deleteInjector = new TopicDeleteInjector(metaStore, topicClient, registryClient);
+
+    when(metaStore.getSource("SOMETHING")).thenAnswer(inv -> source);
+    when(source.isSerdeFormat(DataSourceSerDe.AVRO)).thenReturn(false);
+    when(source.getKafkaTopicName()).thenReturn("something");
+
+    dropStatement = givenStatement(
+        "DROP STREAM SOMETHING", new DropStream(QualifiedName.of("SOMETHING"), false, true));
+  }
+
+  @Test
+  public void shouldDoNothingForNonDropStatements() {
+    // Given:
+    final ConfiguredStatement<ListProperties> listProperties =
+        givenStatement("LIST", new ListProperties(Optional.empty()));
+
+    // When:
+    final ConfiguredStatement<ListProperties> injected = deleteInjector.inject(listProperties);
+
+    // Then:
+    assertThat(injected, is(sameInstance(listProperties)));
+  }
+
+  @Test
+  public void shouldDoNothingIfNoDeleteTopic() {
+    // Given:
+    dropStatement = givenStatement(
```

Comment:
It would probably make the test more readable if all assignments to `dropStatement` were actually replaced with appropriate constants `DROP_WITH_DELETE_TOPIC` & `DROP_WITHOUT_DELETE_TOPIC`:

```java
  @Test
  public void shouldDoNothingIfNoDeleteTopic() {
    // When:
    final ConfiguredStatement<DropStream> injected = deleteInjector.inject(DROP_WITHOUT_DELETE_TOPIC);

     // Then:
    assertThat(injected, is(sameInstance(DROP_WITHOUT_DELETE_TOPIC)));
    verifyZeroInteractions(topicClient, registryClient);  
}
```

Note: I've added a check for zero interactions with `registryClient` too...